src/Components/Courses.js

```diff
@@ -1,18 +1,34 @@
-import React from 'react'
+import React, { useEffect, useContext } from 'react'
 import { useSearchParams } from 'react-router-dom'
+import { getAllCourses } from '../apiCalls'
+import { DiscContext } from '../context'
 import SingleCourse from './SingleCourse'
+import Card from './Card'
 
 const Courses = () => {
   const [ searchParams ] = useSearchParams()
+  const { discContext, setDiscContext } = useContext(DiscContext)
 
-  // const currentCourses = apiCalls.getCourses(searchParams)
+  const currentCourses = discContext.courses.map(course => {
+    console.log(course)
+    return (
+      <Card course={course} key={course.id}/>
+    )
+  })
 
+
+  useEffect(() => {
+    getAllCourses()
+    .then(data =>  {
+      setDiscContext({courses: data})
+    })
+  }, [])
```

Comment:
The empty square brackets is important here. Without it, the useEffect would fire any time there is an udate to any part of the component. By passing in the empty square brackets we are telling React to run only when the values specified in the square brackets are changed, which in this case is nothing. Therefore, this effect will only run when the page loads, keeping us from making a bunch of expensive network requests.